src/cpp/src/tokenizer.cpp

```diff
@@ -335,6 +337,31 @@ class Tokenizer::TokenizerImpl {
         return std::vector<std::string>(res_data, res_data + res.get_shape()[0]);
     }
 
+    std::string patch_chat_template(std::string template_str) {
+        // Replace what jinja2cpp doesn't support
+        std::pair<std::string, std::string> replace_str_map[] = {
+            {"'}", "' }"},
+            {"{'", "{ '"},
+            {".strip()", ""},
+            {"is not none", "is defined"},
+            {"is none", "is undefined"},
+            {"= none", "= undefined"},
+            // TODO: Jinja2Cpp does not support Python-style slicing, e.g. [1:].
+            // If chat template contains such slicing, we replace it with 
+            // a placeholder at the moment.
+            {"messages[1:]", "slice(messages, 1)"},
```

Comment:
What is supposed to be done in the scope of TODO?

Reply:
I forget to remove this TODO when i fixed callback. Will remove it